### PR TITLE
do not fail at undefined flyte phase

### PR DESCRIPTION
**Pull request by ckiosidis**
# Hey, I just made a Pull Request!

## Description

An undefined phase should not be handled as an error
## Checklist for PR author(s)

- [ ] Changes are covered by unit test
- [ ] All tests pass
- [ ] Code coverage check passes
- [ ] Error handling is tested
- [ ] Errors are handled at the appropriate layer
- [ ] Errors that cannot be handled where they occur are propagated
- [ ] (optional) Changes are covered by system test
- [ ] Relevant documentation updated
- [ ] This PR has NO breaking change to public API
- [ ] This PR has breaking change to public API and it is documented

## Checklist for PR reviewer(s)

- [ ] This PR has been incorporated in release note for the coming version
- [ ] Risky changes introduced by this PR have been all considered

### CODE REVIEW

File: styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlytePhaseTest.java [hunk 1]
```diff
@@ -45,7 +45,7 @@ public void testSuccessfulMapping() {
 
   @Test
   public void testUnknownMapping() {
-    assertThrows(NoSuchElementException.class, () -> FlytePhase.fromProto(Execution.WorkflowExecution.Phase.UNDEFINED));
+    assertThrows(NullPointerException.class, () -> FlytePhase.fromProto(Execution.WorkflowExecution.Phase.forNumber(100)));
```
Comment (sonjaer) [inline styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlytePhaseTest.java:48]: What does 100 mean?

File: styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java [hunk 2]
```diff
@@ -118,6 +118,8 @@ void emitFlyteEvents(ExecutionOuterClass.Execution execution, RunState runState)
     final Execution.WorkflowExecution.Phase phase = execution.getClosure().getPhase();
     final FlytePhase flytePhase = FlytePhase.fromProto(phase);
     switch (flytePhase) {
+      case UNDEFINED:
+        LOG.info("Keep polling with FlytePhase UNDEFINED for: " + runState.workflowInstance());
```
Comment (sonjaer) [inline styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java:122]: Would we want to break here?

Comment (ckiosidis) [reply styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java:122]: yes

File: styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java [hunk 3]
```diff
@@ -118,6 +118,9 @@ void emitFlyteEvents(ExecutionOuterClass.Execution execution, RunState runState)
     final Execution.WorkflowExecution.Phase phase = execution.getClosure().getPhase();
     final FlytePhase flytePhase = FlytePhase.fromProto(phase);
     switch (flytePhase) {
+      case UNDEFINED:
```
Comment (sonjaer) [inline styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java:121]: Do we want to test this stage in FlyteAdminClientRunnerTest?

Comment (sonjaer) [reply styx-scheduler-service/src/main/java/com/spotify/styx/flyte/FlyteAdminClientRunner.java:121]: nvm. missed it

File: styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlyteAdminClientRunnerTest.java [hunk 4]
```diff
@@ -283,6 +283,23 @@ public void testEmitFlyteEventsExceptionHandling() throws IsClosedException {
...
+  public void testUndefinedShouldNotThrowException() throws Exception {
+    WorkflowInstance workflowInstance = createWorkflowInstance();
+
+    when(flyteAdminClient.getExecution("flyte-test", "testing", "execution-name")).thenReturn(
+        ExecutionOuterClass.Execution
+            .newBuilder()
+            .setClosure(ExecutionOuterClass.ExecutionClosure.newBuilder()
+                .setPhase(Execution.WorkflowExecution.Phase.UNDEFINED).build())
+            .build());
+    when(runState.workflowInstance()).thenReturn(workflowInstance);
```
Comment (sonjaer) [inline styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlyteAdminClientRunnerTest.java:296]: Why is this needed?

File: styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlyteAdminClientRunnerTest.java [hunk 5]
```diff
@@ -283,6 +283,23 @@ public void testEmitFlyteEventsExceptionHandling() throws IsClosedException {
...
+
+    final FlyteExecutionId flyteExecutionId =
+        FlyteExecutionId.create("flyte-test", "testing", "execution-name");
+    flyteRunner.poll(flyteExecutionId, runState);
```
Comment (sonjaer) [inline styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlyteAdminClientRunnerTest.java:300]: Do we want to verify no interaction with stateManager?

Comment (ckiosidis) [reply styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlyteAdminClientRunnerTest.java:300]: good idea

File: styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlytePhaseTest.java (hunk 1 above)
Comment (ckiosidis) [reply styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlytePhaseTest.java]: i thought it would work but it won't. it supposed to pick up the n-th item from the enum, but it just throws a NPE

File: styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlyteAdminClientRunnerTest.java (hunk 4 above)
Comment (ckiosidis) [reply styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlyteAdminClientRunnerTest.java:296]: because the logline inside the emitFlyteEvents calls this method

File: styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlyteAdminClientRunnerTest.java [hunk 6]
```diff
@@ -283,6 +284,25 @@ public void testEmitFlyteEventsExceptionHandling() throws IsClosedException {
                 Execution.WorkflowExecution.Phase.SUCCEEDED).build()).build(), runState));
   }
 
+  @Test
+  public void testUndefinedShouldNotThrowException() throws Exception {
```
Comment (sonjaer) [inline styx-scheduler-service/src/test/java/com/spotify/styx/flyte/FlyteAdminClientRunnerTest.java:288]: could name this something else, testUndefinedShouldDoNothing or testUndefinedShouldPollAgain, but I am fine with naming also.

**Review by sonjaer — APPROVED** [review]
